src/Stryker.Core/Stryker.Core/Baseline/AzureFileShareBaselineProvider.cs

```diff
@@ -0,0 +1,212 @@
+            {
+                var succesfullyResizedFile = await ResizeFile(byteSize, version);
+
+                if (!succesfullyResizedFile)
+                {
+                    return;
+                }
+            }
+
+
+            await UploadFile(reportJson, byteSize, version);
+        }
+
+        public async Task<bool> CreateDirectory(string version)
+        {
+            var subdir = $"strykerOutput/baselines/{version}";
+            _logger.LogInformation("Creating directory {0}", subdir);
+
+            var url = new Uri($"{_options.AzureFileStorageUrl}/{subdir}?restype=directory&sv={_options.AzureSAS}");
+
+            var requestMessage = new HttpRequestMessage(HttpMethod.Put, url);
+
+            SetWritingHeaders(requestMessage);
+
+            var response = await _httpClient.SendAsync(requestMessage);
+
+            if (response.StatusCode == HttpStatusCode.Created)
+            {
+                _logger.LogDebug("Succesfully created directory {0} on fileshare {1}", subdir, _options.AzureFileStorageUrl);
+                return true;
+            }
+            else if (response.StatusCode == HttpStatusCode.Conflict)
+            {
+                _logger.LogDebug("Directory {0} already excists on fileshare {1}", subdir, _options.AzureFileStorageUrl);
+                return true;
+            }
+
+            _logger.LogError("Creating directory failed with status {0} and message {1}", response.StatusCode.ToString(), response.RequestMessage);
+            return false;
+
+        }
+
+        private async Task<bool> AllocateFileLocation(int byteSize, string version)
+        {
+            _logger.LogInformation("Allocating storage for file");
+
+            var subdir = $"strykerOutput/baselines/{version}";
+            var filename = "stryker-report.json";
+            var url = new Uri($"{_options.AzureFileStorageUrl}/{subdir}/{filename}?sv={_options.AzureSAS}");
+
+            var requestMessage = new HttpRequestMessage(HttpMethod.Put, url);
+
+            SetWritingHeaders(requestMessage);
+
+            requestMessage.Headers.Add("x-ms-type", "file");
+            requestMessage.Headers.Add("x-ms-content-length", byteSize.ToString());
+
+
+            var response = await _httpClient.SendAsync(requestMessage);
+
+            if (response.StatusCode == HttpStatusCode.Created)
+            {
+                _logger.LogDebug("Succesfully allocated storage on fileshare {0}", _options.AzureFileStorageUrl);
+                return true;
+            }
+            else
+            {
+                _logger.LogError("Azure File Storage upload failed with statuscode {0} and message: {1}", response.StatusCode.ToString(), await response.Content.ReadAsStringAsync());
+                return false;
+            }
+        }
+
+        private async Task UploadFile(string report, int byteSize, string version)
+        {
+            _logger.LogInformation("Uploading file to azure file storage");
```

Comment:
This doesn't need to be logged here and not as information. Place it as debug logging before the SendAsync and add the url of the file share without sas.